src/runtimes/node/bundler.js

```diff
@@ -26,17 +27,21 @@ const supportsAsyncAPI = semver.satisfies(process.version, '>=9.x')
 // eslint-disable-next-line node/no-sync
 const buildFunction = supportsAsyncAPI ? esbuild.build : esbuild.buildSync
 
+// eslint-disable-next-line max-statements
 const bundleJsFile = async function ({
   additionalModulePaths,
   basePath,
   config,
-  destFolder,
   externalModules = [],
   ignoredModules = [],
   name,
   srcDir,
   srcFile,
 }) {
+  // We use a temporary directory as the destination for esbuild files to avoid
+  // any naming conflicts with files generated by other functions.
+  const targetDirectory = await tmpName()
```

Comment:
Just double-checking: are we creating `targetDirectory` intermediary directories if they do not exist?
If not, `tmp-promise` `dir()` method can be used for that purpose.

Reply:
> Just double-checking: are we creating `targetDirectory` intermediary directories if they do not exist?
> If not, `tmp-promise` `dir()` method can be used for that purpose.

Yes, esbuild takes care of creating all the intermediary directories, so I don't think we need to do that ourselves here.